source/mqtt/Mqtt5Client.cpp

```diff
@@ -233,6 +247,36 @@ namespace Aws
 
             Mqtt5ClientOptions::~Mqtt5ClientOptions() {}
 
+            Mqtt5to3AdapterOptions *Mqtt5ClientOptions::NewMqtt5to3AdapterOptions() const noexcept
+            {
+                Mqtt5to3AdapterOptions *adapterOptions = new Mqtt5to3AdapterOptions();
```

Comment:
Debatable: If `m_mqtt5to3AdapterOptions` is populated directly in this method, then there will be no need in allocating `m_mqtt5to3AdapterOptions` on the heap or even making it a pointer.
My main concern is a `new` call in the code: there is no issue with it, but it always enforces you to make sure that corresponding `delete` will be called. Maybe at least replace it with `std::unique_ptr`?

Reply:
In the crt we always pair `*_destroy()` for anything that has a `*_new()` that takes an allocator to make sure we clean up anything we allocate. Unsure if we have the same for something with `New` in cpp but I feel like I've seen `New` used that returns a pointer or a null with no corresponding `delete` or `destroy`. Good to stay consistent though so worth investigating.